stravalib/model.py

```diff
@@ -759,20 +1043,23 @@ class Activity(
     end_latlng: Optional[LatLon] = None
     map: Optional[Map] = None
     gear: Optional[Gear] = None
-    best_efforts: Optional[List[BestEffort]] = None
-    segment_efforts: Optional[List[SegmentEffort]] = None
-    splits_metric: Optional[List[Split]] = None
-    splits_standard: Optional[List[Split]] = None
+    best_efforts: Optional[List[DetailedSegmentEffort]] = None
+    segment_efforts: Optional[List[DetailedSegmentEffort]] = None
+    # Ignoring types here given there are overrides
+    splits_metric: Optional[List[Split]] = None  # type: ignore
+    splits_standard: Optional[List[Split]] = None  # type: ignore
     photos: Optional[ActivityPhotoMeta] = None
-    laps: Optional[List[ActivityLap]] = None
+    # TODO Detailed activity has this defined as - Optional[List[Lap]]
+    laps: Optional[List[Union[ActivityLap, Lap]]] = None
 
     # Added for backward compatibility
     # TODO maybe deprecate?
-    TYPES: ClassVar[Tuple] = get_args(
+    TYPES: ClassVar[Tuple[Any, ...]] = get_args(
```

Comment:
It sounds like a good idea to deprecate those they also seems quite intertwined with some pydantic v1 magic that might cause problem migration to v2

Reply:
cool - i just opened an issue here:

https://github.com/stravalib/stravalib/issues/400

i know that the migration to v2 will potentially be significant so let's implement this fix in a separate (smaller!) pr